src/renderer/vt/Xterm256Engine.cpp

```diff
@@ -42,11 +43,77 @@ Xterm256Engine::Xterm256Engine(_In_ wil::unique_hfile hPipe,
     // We have to do this here, instead of in PaintBufferGridLines, because
     //      we'll have already painted the text by the time PaintBufferGridLines
     //      is called.
+    // TODO:GH#2915 Treat underline separately from LVB_UNDERSCORE
     RETURN_IF_FAILED(_UpdateUnderline(legacyColorAttribute));
 
+    // Only do extended attributes in xterm-256color, as to not break telnet.exe.
+    RETURN_IF_FAILED(_UpdateExtendedAttrs(extendedAttrs));
+
     return VtEngine::_RgbUpdateDrawingBrushes(colorForeground,
                                               colorBackground,
-                                              isBold,
+                                              WI_IsFlagSet(extendedAttrs, ExtendedAttributes::Bold),
                                               _ColorTable,
                                               _cColorTable);
 }
+
+// Routine Description:
+// - Write a VT sequence to either start or stop underlining text.
+// Arguments:
+// - legacyColorAttribute: A console attributes bit field containing information
+//      about the underlining state of the text.
+// Return Value:
+// - S_OK if we succeeded, else an appropriate HRESULT for failing to allocate or write.
+[[nodiscard]] HRESULT Xterm256Engine::_UpdateExtendedAttrs(const ExtendedAttributes extendedAttrs) noexcept
+{
+    // Helper lambda to check if a state (attr) has changed since it's last
+    // value (lastState), and appropriately start/end that state with the given
+    // begin/end functions.
+    auto updateFlagAndState = [extendedAttrs](const ExtendedAttributes attr,
+                                              bool& lastState,
+                                              std::function<HRESULT(void)> beginFn,
+                                              std::function<HRESULT(void)> endFn) -> HRESULT {
+        // Unfortunately, we can't use WI_IsFlagSet here, since that requires a
+        // compile-time check that the flag (attr) is a _single_ flag, and that
+        // won't work in this case.
+        const bool flagSet = (extendedAttrs & attr) == attr;
+        if (flagSet != lastState)
+        {
+            if (flagSet)
+            {
+                RETURN_IF_FAILED(beginFn());
+            }
+            else
+            {
+                RETURN_IF_FAILED(endFn());
+            }
+            lastState = flagSet;
+        }
+        return S_OK;
+    };
+
+    auto hr = updateFlagAndState(ExtendedAttributes::Italics,
+                                 _usingItalics,
+                                 std::bind(&Xterm256Engine::_BeginItalics, this),

```

Comment:
binds are sooooooooo expensive. :|

Reply:
i wonder if there's a trick you can do with auto-typed args in lambdas

Reply:
If we have a particular qualm with `std::bind`, we should probably file a code health work item with the proposed replacement pattern for them. There's a ton of existing ones.

Reply:
I just did a cursory find-all for `std::bind`, and it didn't look like there were any other perf-intensive (loops) places where we did a bind, usually only on startup. @DHowett-MSFT think we should still file a follow-up? I was able to remove this one at least.